scripts/casefold_db.py

```diff
@@ -137,9 +132,54 @@ def update_local_associations(
         # We've finished updating the database, committing the transaction.
         db.commit()
 
+    # iterate through the mxids and send emails
+    if send_email and not dry_run:
```

Comment:
It's not immediately obvious to me that we want to delete the emails from the DB *first* and *then* send the emails? It means that if you run this script and it fails at sending the emails (e.g. because of a temporary failure), rerunning it means we won't send *any* emails.

I think perhaps the correct thing to do here is to individually delete the entries for an mxid after we've sent the email?

Reply:
Yeah after reflecting on it that's probably the best way to do it.

Reply:
I've pushed https://github.com/matrix-org/sydent/pull/379/commits/e3deaefdc044fffc4ccfc9b1b8a0090cbacf7fe9 that does this.